atomium/includes/common.inc

```diff
@@ -675,33 +675,141 @@ function _atomium_array_flatten(array $array, array &$storage, $parentKey = '')
  */
 function atomium_get_settings($setting_keys, $base_themes = TRUE, array $value_callbacks = array('trim')) {
   $theme_key = $GLOBALS['theme_key'];
+  $settings_flat_filtered = _atomium_get_settings_flat_filtered($theme_key, $base_themes, $value_callbacks);
 
-  $settings = array();
+  return isset($settings_flat_filtered[$setting_keys])
+    ? $settings_flat_filtered[$setting_keys]
+    : array();
+}
 
-  if ($theme_settings = atomium_get_theme_info($theme_key, 'settings', $base_themes)) {
-    _atomium_array_flatten(
-      $theme_settings,
-      $settings
-    );
+/**
+ * Get the settings in a flat and filtered array.
+ *
+ * @param string $theme_key
+ *   The theme.
+ * @param bool $base_themes
+ *   True to inherit settings from base themes, false otherwise.
+ * @param array $value_callbacks
+ *   Callbacks to apply to setting values.
+ *
+ * @return mixed
+ *   The settings.
+ *
+ * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+ * @SuppressWarnings(PHPMD.NPathComplexity)
+ */
+function _atomium_get_settings_flat_filtered($theme_key, $base_themes = TRUE, array $value_callbacks = array('trim')) {
+  static $settings_flat_filtered_by_cid = array();
```

Comment:
Why not use drupal_static?

Reply:
The benefit of `drupal_static()` would be that other code (e.g. unit tests) would be able to reset this cache. Is someone planning to do this?

If we use `drupal_static()`, we should also use the drupal_static_fast() pattern I guess..